File: 151. Reverse Words in a String/_151.py
class Solution:
    def reverseWords(self, s: str) -> str:
        res = s.split(' ')
        res.reverse()
        s = ""
        res1 = list()
        for i in res:
            if i == "" or i == " ":
                continue
            res1.append(i)
        return " ".join(res1)

    def reverseWords1(self, s: str) -> str:
        return ' '.join(s.strip().split()[::-1])

    def reverseWords2(self, s):
        word = ""
        words = ""
        s = s[::-1]
        for j, i in enumerate(s):
            if i != " " and word != "" and s[j-1] == " ":
                words += (word + " ")
                word = i
            elif i != " ":
                word = i + word
            else:
                continue
        words += word
        return(words)

a = Solution()
print(a.reverseWords1( "a good   example"))

File: 151. Reverse Words in a String/151.go
package leetcode

import(
	"strings"
)

func reverseWords(s string) string {
	res := strings.Split(s, " ")
	res1 := make([]string, 0)
	for i:=len(res)-1; i>=0; i--{
		if res[i] == "" || res[i] == " "{
			continue
		}
		res1 = append(res1, res[i])
	}
	return strings.Join(res1, " ")
}

func reverse(m *[]string, i int, j int){
	for {
        if i > j {
			break
		}
		(*m)[i], (*m)[j] = (*m)[j], (*m)[i]
		i++
		j--
	}
}

func reverseWords1(s string) string {
	ss:=strings.Fields(s)
	reverse(&ss,0,len(ss)-1)
    return strings.Join(ss," ")
}